platform/core.network/src/org/netbeans/core/network/utils/IpAddressUtils.java

```diff
@@ -52,7 +53,18 @@
     private static final Pattern IPV4_PATTERN = Pattern.compile("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
     private static final RequestProcessor RP = new RequestProcessor("DNSBackgroundResolvers", 10);
 
-    private IpAddressUtils() {}
+    IpAddressUtils() {}
+
+    private static IpAddressUtils INSTANCE;
+    private static synchronized IpAddressUtils getDefault() {
+        if (INSTANCE == null) {
+            INSTANCE = Lookup.getDefault().lookup(IpAddressUtils.class);
```

Comment:
This makes `IpAddressUtils` mockable and thus it avoids the reflection based tricks with `NameService` that are broken on JDK11.